### ISSUE TITLE
Duplicate file warning from dependencies when building über-jar 

### ISSUE THREAD

**Issue by RolKau** [quarkusio/quarkus#41701]
### Describe the bug

The minimal reproducing example attached as pom.xml in the Additional Information section produces a warning when building.

### Expected behavior

 A build consisting of nothing but core components should proceed without warnings.

### Actual behavior

When building the minimal example, the following warning is issued:

    [WARNING] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Dependencies with duplicate files detected. The dependencies [org.jctools:jctools-core::jar:4.0.3[paths: .m2/repository/org/jctools/jctools-core/4.0.3/jctools-core-4.0.3.jar;], io.netty:netty-common::jar:4.1.108.Final[paths: .m2/repository/io/netty/netty-common/4.1.108.Final/netty-common-4.1.108.Final.jar;]] contain duplicate files, e.g. META-INF/maven/org.jctools/jctools-core/pom.xml

### How to Reproduce?

1. Create a new, empty directory
2. Save the attached file in the Additional Information as `pom.xml`
3. Run the command `mvn package`

### Output of `uname -a` or `ver`

Linux 5.15.0-107-generic #117~20.04.1-Ubuntu SMP Tue Apr 30 10:35:57 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "17.0.11" 2024-04-16 OpenJDK Runtime Environment (build 17.0.11+9-Ubuntu-120.04.2) OpenJDK 64-Bit Server VM (build 17.0.11+9-Ubuntu-120.04.2, mixed mode, sharing)

### Quarkus version or git rev

3.12.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.4 (dfbb324ad4a7c8fb0bf182e6d91b0ae20e3d2dd9) Maven home: /usr/local/maven Java version: 17.0.11, vendor: Ubuntu, runtime: /usr/lib/jvm/java-17-openjdk-amd64 Default locale: en_IE, platform encoding: UTF-8 OS name: "linux", version: "5.15.0-107-generic", arch: "amd64", family: "unix"

### Additional information

[pom.xml.txt](https://github.com/user-attachments/files/16102660/pom.xml.txt)

**Comment by geoand** [comment]
cc @franz1981

**Comment by franz1981** [comment]
@jponge probably knows

**Comment by gsmet** [comment]
I noticed the issue a while back and created https://github.com/quarkusio/quarkus/pull/40048 but @aloubyansky wasn't happy about it :).

**Comment by jponge** [comment]
That's because Netty shades JCTools, while Mutiny brings a regular dependency.

**Comment by aloubyansky** [comment]
Right. Looking at how it's shaded, the package is changed, so the classes aren't overridden. In this case we could probably suppress the warning. If the package wasn't changed, I think it'd be worth bringing attention to the fact that contents of different versions of the same library are getting merged.
Although `META-INF/maven` could potentially be used by other tools to check which packages were merged. That could be addressed by generating and distributing an SBOM though.

**Comment by maxandersen** [comment]
I'm hitting this now too building quarkiverse/quarkus-mcp-servers filesystem module which has no deps outside quarkus itself afaik - any suggested solution?

**Comment by maxandersen** [comment]
seem i only hit this when using graalvm 21.0.1 ce edition....did it fix itself?

**Comment by geoand** [comment]
Is this still an issue with Quarkus 3.32?

**Comment by geoand** [comment]
Closing for lack of feedback

**Comment by RolKau** [comment]
Sorry, I didn't see the original call for testing with Quarkus 3.32. The answer to that question is: I don't know if this problem still applies, because the example no longer builds! Running `mvn package` on `pom.xml` attached to this issue now returns the following error:

    [ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:3.32.3:build (default) on project warningsinbuild: Execution default of goal io.quarkus:quarkus-maven-plugin:3.32.3:build failed: Unable to load the mojo 'build' (or one of its required components) from the plugin 'io.quarkus:quarkus-maven-plugin:3.32.3': com.google.inject.ProvisionException: Unable to provision, see the following errors: 
    [ERROR] 1) No implementation for io.quarkus.maven.QuarkusBootstrapProvider was bound.

**Comment by aloubyansky** [comment]
@RolKau that seems like an old and not any more supported Maven version in that version of Quarkus.

**Comment by RolKau** [comment]
Indeed; building in a newer Debian (Trixie) environment with now Maven v3.9.9, the build now proceeds, with the reported warning with Quarkus v3.12.1, but with the warning gone with v3.32.3. So the issue could be closed with "evaporated over time" or something like that instead of "not planned". (Hat tip to @aloubyansky for troubleshooting).

**Comment by geoand** [comment]
👍🏽
